### ISSUE TITLE
Experiments with MNIST

### ISSUE THREAD

**Issue by diff7** [jackyzyb/bayesian-coresets-optimization#3]
Hi, I was wondering if I could reduce data-set size using your method. 
I used examples in *reinman_poisson_regression* to build a corset and then I was fitting a simple sklearn logistic regression. 
These are results I have got. There is a  chance that I misunderstood something about the method. Or probably this method is not applicable for images. Could comment it? Thank you!

![image](https://user-images.githubusercontent.com/34035501/106883744-7326b480-66f1-11eb-8760-bebdaa86d213.png)

**Comment by jackyzyb** [comment]
Hi, thanks for sharing your interesting results! Sorry I was not aware of your post for this long... 

My guess for your results is: as IHT does a better job in optimization, it might cause overfitting if the number MC samples, or equivalently the the dimension of the vector g (see eq. 3 in our paper), is much smaller than the number of data. 

There is a simple way to verify whether it is due to overfitting: since IHT and GIGA use the same objective function (the l2 loss), one could check their l2 loss to see the quality of optimization. However, good optimization does not necessarily lead to good performance. Therefore, one could use a validation set of MC samples independent from the training samples to see whether there is overfitting. 

If so, there are some techniques to overcome overfitting. For example, increase the number of MC samples, use the validation set to early stop the optimization, etc. 

Hope this help!

**Comment by diff7** [comment]
Hi, thank you for your reply. I will try to make it work!
